src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +93,13 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex() == Person.Sex.WOMAN
+                        && person.getAge() >= femaleAge)
+                .map(person -> person.getCats())
+                .flatMap(cats -> cats.stream())
```

Comment:
```suggestion
                .flatMap(p -> p.getCatList().stream())
```

we can simplify this